coreapps-web-2.2/src/main/java/org/openmrs/module/coreapps/diagnosisCompatibility/DiagnosisServiceCompatibility2_2.java

```diff
@@ -0,0 +1,31 @@
+package org.openmrs.module.coreapps.diagnosisCompatibility;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.api.DiagnosisService;
+import org.openmrs.ui.framework.SimpleObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DiagnosisServiceCompatibility2_2   {
+	
+	@Autowired
+	@Qualifier("diagnosisService")
+	private DiagnosisService diagnosisService;
+	
+	
+	public List<SimpleObject> getPrimaryDiagnosesAsSimpleObjects(Encounter encounter) throws Exception {
+		List<Diagnosis> list = diagnosisService.getDiagnoses(encounter);
+		List<SimpleObject> results = new ArrayList<SimpleObject>(list.size());
+		for (Diagnosis diagnosis : list) {
+			SimpleObject simpleObject = SimpleObject.create(diagnosis);
```

Comment:
@mozzy11 @ibacher @dkayiwa  could this be the right way of passing on a diagnosis object to SimpleObject,

Reply:
No... you really need to add the properties to the `SimpleObject` that you need. The `parseObsAsDiagnosis()` method should show you how this is done with Obs. We'll need similar logic, but for the existing Diagnosis object.